libiop/bcs/common_bcs_parameters.tcc

```diff
@@ -15,10 +15,10 @@ bcs_transformation_parameters<FieldT, MT_root_hash> default_bcs_params(
     params.hash_enum = hash_type;
     /* TODO: Push setting leaf hash into internal BCS code. Currently 2 is fine, as leaf size is internally unused. */
     const size_t leaf_size = 2;
-    params.leafhasher_ = get_leafhash<FieldT, MT_root_hash>(hash_type, security_parameter, leaf_size);
+    params.leafhasher_ = get_leafhash<MT_root_hash, FieldT>(hash_type, security_parameter, leaf_size);
```

Comment:
Was this just a bug before?

Reply:
No, I switched the order to be consistent with similar functions

Reply:
What was here before was right I believe https://github.com/scipr-lab/libiop/blob/master/libiop/bcs/hashing/hash_enum.hpp#L33-L34

Reply:
I still believe the template ordering before was right

Reply:
Oh I see, you switched the template ordering of the internal command. Currently there are inconsistent orderings throughout, I believe the old `<FieldT, MT_root_hash>` was consistent here. We should consistently be using one or the other, I don't really have a preference for which one.

Reply:
I switched all instances of the leaf hash functions to be the same as the hashchain.